{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5748b653-9a09-48c2-a94d-c9ad44e678f4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.chdir('/Users/akki/Desktop/USC/FreshmanYear/QBIO490/qbio_490_AkanshaSallakonda/analysis_data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e2f24393-af91-4d71-b818-3a43f8c00a70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Description</th>\n",
       "      <th>Data reuse status</th>\n",
       "      <th>Publication link</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dataset name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Brca</th>\n",
       "      <td>breast cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33212010/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ccrcc</th>\n",
       "      <td>clear cell renal cell carcinoma (kidney)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31675502/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon</th>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/31031003/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endometrial</th>\n",
       "      <td>endometrial carcinoma (uterine)</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32059776/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gbm</th>\n",
       "      <td>glioblastoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33577785/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hnscc</th>\n",
       "      <td>head and neck squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/33417831/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lscc</th>\n",
       "      <td>lung squamous cell carcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34358469/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Luad</th>\n",
       "      <td>lung adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/32649874/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovarian</th>\n",
       "      <td>high grade serous ovarian cancer</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/27372738/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pdac</th>\n",
       "      <td>pancreatic ductal adenocarcinoma</td>\n",
       "      <td>no restrictions</td>\n",
       "      <td>https://pubmed.ncbi.nlm.nih.gov/34534465/</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UcecConf</th>\n",
       "      <td>endometrial confirmatory carcinoma</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GbmConf</th>\n",
       "      <td>glioblastoma confirmatory</td>\n",
       "      <td>password access only</td>\n",
       "      <td>unpublished</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           Description     Data reuse status  \\\n",
       "Dataset name                                                                   \n",
       "Brca                                     breast cancer       no restrictions   \n",
       "Ccrcc         clear cell renal cell carcinoma (kidney)       no restrictions   \n",
       "Colon                                colorectal cancer       no restrictions   \n",
       "Endometrial            endometrial carcinoma (uterine)       no restrictions   \n",
       "Gbm                                       glioblastoma       no restrictions   \n",
       "Hnscc            head and neck squamous cell carcinoma       no restrictions   \n",
       "Lscc                      lung squamous cell carcinoma       no restrictions   \n",
       "Luad                               lung adenocarcinoma       no restrictions   \n",
       "Ovarian               high grade serous ovarian cancer       no restrictions   \n",
       "Pdac                  pancreatic ductal adenocarcinoma       no restrictions   \n",
       "UcecConf            endometrial confirmatory carcinoma  password access only   \n",
       "GbmConf                      glioblastoma confirmatory  password access only   \n",
       "\n",
       "                                       Publication link  \n",
       "Dataset name                                             \n",
       "Brca          https://pubmed.ncbi.nlm.nih.gov/33212010/  \n",
       "Ccrcc         https://pubmed.ncbi.nlm.nih.gov/31675502/  \n",
       "Colon         https://pubmed.ncbi.nlm.nih.gov/31031003/  \n",
       "Endometrial   https://pubmed.ncbi.nlm.nih.gov/32059776/  \n",
       "Gbm           https://pubmed.ncbi.nlm.nih.gov/33577785/  \n",
       "Hnscc         https://pubmed.ncbi.nlm.nih.gov/33417831/  \n",
       "Lscc          https://pubmed.ncbi.nlm.nih.gov/34358469/  \n",
       "Luad          https://pubmed.ncbi.nlm.nih.gov/32649874/  \n",
       "Ovarian       https://pubmed.ncbi.nlm.nih.gov/27372738/  \n",
       "Pdac          https://pubmed.ncbi.nlm.nih.gov/34534465/  \n",
       "UcecConf                                    unpublished  \n",
       "GbmConf                                     unpublished  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (1.1.2) is out-of-date. Latest is 1.5.7. Please run 'pip install --upgrade cptac' to update it. (/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/threading.py, line 917)\n"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "cptac.list_datasets()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cbb84809-a5f1-4da6-ad8a-6cabb5cc24a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Below are the dataframes contained in this dataset and their dimensions:\n",
      "\n",
      "clinical\n",
      "\t194 rows\n",
      "\t171 columns\n",
      "CNV\n",
      "\t110 rows\n",
      "\t19285 columns\n",
      "followup\n",
      "\t352 rows\n",
      "\t27 columns\n",
      "medical_history\n",
      "\t370 rows\n",
      "\t4 columns\n",
      "methylation\n",
      "\t107 rows\n",
      "\t15885 columns\n",
      "phosphoproteomics\n",
      "\t194 rows\n",
      "\t81550 columns\n",
      "phosphoproteomics_gene\n",
      "\t194 rows\n",
      "\t6127 columns\n",
      "proteomics\n",
      "\t194 rows\n",
      "\t11710 columns\n",
      "somatic_mutation\n",
      "\t8350 rows\n",
      "\t3 columns\n",
      "transcriptomics\n",
      "\t185 rows\n",
      "\t19275 columns\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "cptac.download(dataset = \"Ccrcc\")\n",
    "\n",
    "ccrcc = cptac.Ccrcc()\n",
    "ccrcc.list_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "18417206-e73f-4a18-8835-d8e1cfa20d0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#masking to only have Stage I-III tumors\n",
    "\n",
    "clinical_data = ccrcc.get_clinical()\n",
    "stage_mask = np.where(clinical_data.tumor_stage_pathological != \"Stage IV\", True, False)\n",
    "clinical_data_mid = clinical_data[stage_mask]\n",
    "clinical_data = clinical_data_mid[clinical_data_mid.tumor_stage_pathological.notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "34ff6ac0-a1a1-4e95-ab00-e296b2cc0820",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/pandas/core/internals/blocks.py:366: RuntimeWarning: divide by zero encountered in log2\n",
      "  result = func(self.values, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "#Masking RNA data to reflect what data is available in clinical data\n",
    "rna_data = ccrcc.get_transcriptomics()\n",
    "rna_mask = rna_data.index.isin(clinical_data.index)\n",
    "rna_data = rna_data[rna_mask]\n",
    "rna_data = np.log2(rna_data) \n",
    "rna_data = rna_data.dropna(axis=1)\n",
    "rna_data = rna_data.T.loc[~rna_data.T.index.duplicated(), :].T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "3872549d-a2e5-490b-b331-89c422e46ac7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Masking protein data to reflect what data is available in clinical data\n",
    "protein_data = ccrcc.get_proteomics()\n",
    "protein_data.columns = protein_data.columns.get_level_values(0) \n",
    "protein_mask = protein_data.index.isin(clinical_data.index)\n",
    "protein_data = protein_data[protein_mask]\n",
    "protein_data = protein_data.dropna(axis=1)\n",
    "protein_data = protein_data.T.loc[~protein_data.T.index.duplicated(), :].T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "ba987fa1-c1ca-40b6-8b0e-5347fd92fd94",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name\n",
      "MT1H    2.844733\n",
      "MT1G    2.590020\n",
      "MT1F    2.432565\n",
      "MT1E    2.283102\n",
      "MT1B    2.145197\n",
      "dtype: float64\n",
      "Name\n",
      "MMP8       inf\n",
      "SLC6A11    inf\n",
      "SLC5A11    inf\n",
      "IGFL2      inf\n",
      "SLC5A5     inf\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "rna_mean = rna_data.mean()\n",
    "protein_mean = protein_data.mean()\n",
    "\n",
    "top_five_protein = np.absolute(protein_mean).sort_values(ascending=False).head(5)\n",
    "print(top_five_protein)\n",
    "\n",
    "top_five_RNA = np.absolute(rna_mean).sort_values(ascending=False).head(5)\n",
    "print(top_five_RNA)\n",
    "\n",
    "top_RNA_list = [\"NUP98\", \"NMT1\", \"RAB8A\", \"TNPO3\", \"ZNF664\"]\n",
    "top_protein_list = [\"MT1H\", \"MT1G\", \"MT1F\", \"MT1E\", \"MT1B\"]\n",
    "\n",
    "top_RNA_df = rna_data.loc[:, top_RNA_list]\n",
    "top_protein_df = protein_data.loc[:, top_protein_list]\n",
    "\n",
    "#Top 5 differentially expressed RNA genes : MMP8, SLC6A11, SLC5A11, IGFL2, SLC5A5\n",
    "#Top 5 differentially expressed Proteins: MT1H, MT1G, MT1F, MT1E, MT1B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "b9f3d689-04e7-4bdd-acfa-96cb37d30dc2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>NUP98</th>\n",
       "      <th>NMT1</th>\n",
       "      <th>RAB8A</th>\n",
       "      <th>TNPO3</th>\n",
       "      <th>ZNF664</th>\n",
       "      <th>MT1H</th>\n",
       "      <th>MT1G</th>\n",
       "      <th>MT1F</th>\n",
       "      <th>MT1E</th>\n",
       "      <th>MT1B</th>\n",
       "      <th>tumor_stage_pathological</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>4.475587</td>\n",
       "      <td>4.709943</td>\n",
       "      <td>4.511866</td>\n",
       "      <td>4.538224</td>\n",
       "      <td>4.381721</td>\n",
       "      <td>-2.675872</td>\n",
       "      <td>-2.253525</td>\n",
       "      <td>-2.546547</td>\n",
       "      <td>-2.557182</td>\n",
       "      <td>-1.939419</td>\n",
       "      <td>Stage III</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>4.099825</td>\n",
       "      <td>4.505286</td>\n",
       "      <td>4.568616</td>\n",
       "      <td>4.466137</td>\n",
       "      <td>4.387715</td>\n",
       "      <td>-4.050571</td>\n",
       "      <td>-4.060004</td>\n",
       "      <td>-2.921551</td>\n",
       "      <td>-3.591516</td>\n",
       "      <td>-3.363261</td>\n",
       "      <td>Stage I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>4.100547</td>\n",
       "      <td>4.361278</td>\n",
       "      <td>4.739195</td>\n",
       "      <td>4.825138</td>\n",
       "      <td>4.246069</td>\n",
       "      <td>-2.607363</td>\n",
       "      <td>-2.475965</td>\n",
       "      <td>-2.492909</td>\n",
       "      <td>-2.004049</td>\n",
       "      <td>-1.809835</td>\n",
       "      <td>Stage I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>4.391514</td>\n",
       "      <td>4.332005</td>\n",
       "      <td>4.433089</td>\n",
       "      <td>4.594001</td>\n",
       "      <td>4.313825</td>\n",
       "      <td>-1.735521</td>\n",
       "      <td>-2.525792</td>\n",
       "      <td>-3.036604</td>\n",
       "      <td>-2.514611</td>\n",
       "      <td>-1.763428</td>\n",
       "      <td>Stage III</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00088</th>\n",
       "      <td>4.320449</td>\n",
       "      <td>4.248517</td>\n",
       "      <td>4.490979</td>\n",
       "      <td>4.416537</td>\n",
       "      <td>4.265659</td>\n",
       "      <td>-2.404961</td>\n",
       "      <td>-2.110608</td>\n",
       "      <td>-1.972565</td>\n",
       "      <td>-2.471242</td>\n",
       "      <td>-1.731799</td>\n",
       "      <td>Stage III</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646</th>\n",
       "      <td>4.128605</td>\n",
       "      <td>4.685423</td>\n",
       "      <td>4.673625</td>\n",
       "      <td>4.595805</td>\n",
       "      <td>4.921863</td>\n",
       "      <td>-2.183352</td>\n",
       "      <td>-2.047085</td>\n",
       "      <td>-2.162791</td>\n",
       "      <td>-2.431460</td>\n",
       "      <td>-1.742155</td>\n",
       "      <td>Stage III</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648</th>\n",
       "      <td>4.523754</td>\n",
       "      <td>4.319720</td>\n",
       "      <td>5.140829</td>\n",
       "      <td>4.569094</td>\n",
       "      <td>4.142063</td>\n",
       "      <td>-1.454984</td>\n",
       "      <td>-1.196218</td>\n",
       "      <td>-1.313544</td>\n",
       "      <td>-1.182120</td>\n",
       "      <td>-1.103926</td>\n",
       "      <td>Stage II</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649</th>\n",
       "      <td>4.597990</td>\n",
       "      <td>4.047711</td>\n",
       "      <td>4.184536</td>\n",
       "      <td>4.636023</td>\n",
       "      <td>4.694756</td>\n",
       "      <td>-2.214753</td>\n",
       "      <td>-2.534672</td>\n",
       "      <td>-2.485910</td>\n",
       "      <td>-2.172642</td>\n",
       "      <td>-2.360767</td>\n",
       "      <td>Stage III</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651</th>\n",
       "      <td>4.241089</td>\n",
       "      <td>4.465306</td>\n",
       "      <td>4.823650</td>\n",
       "      <td>4.909012</td>\n",
       "      <td>4.214024</td>\n",
       "      <td>-0.137019</td>\n",
       "      <td>-0.360000</td>\n",
       "      <td>-0.295140</td>\n",
       "      <td>-3.646967</td>\n",
       "      <td>-0.165253</td>\n",
       "      <td>Stage II</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808</th>\n",
       "      <td>4.476363</td>\n",
       "      <td>4.189120</td>\n",
       "      <td>4.141382</td>\n",
       "      <td>4.853221</td>\n",
       "      <td>4.731376</td>\n",
       "      <td>-1.986306</td>\n",
       "      <td>-1.697919</td>\n",
       "      <td>-2.052849</td>\n",
       "      <td>-1.963289</td>\n",
       "      <td>-1.544815</td>\n",
       "      <td>Stage I</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name           NUP98      NMT1     RAB8A     TNPO3    ZNF664      MT1H  \\\n",
       "Patient_ID                                                               \n",
       "C3L-00004   4.475587  4.709943  4.511866  4.538224  4.381721 -2.675872   \n",
       "C3L-00010   4.099825  4.505286  4.568616  4.466137  4.387715 -4.050571   \n",
       "C3L-00026   4.100547  4.361278  4.739195  4.825138  4.246069 -2.607363   \n",
       "C3L-00079   4.391514  4.332005  4.433089  4.594001  4.313825 -1.735521   \n",
       "C3L-00088   4.320449  4.248517  4.490979  4.416537  4.265659 -2.404961   \n",
       "...              ...       ...       ...       ...       ...       ...   \n",
       "C3N-01646   4.128605  4.685423  4.673625  4.595805  4.921863 -2.183352   \n",
       "C3N-01648   4.523754  4.319720  5.140829  4.569094  4.142063 -1.454984   \n",
       "C3N-01649   4.597990  4.047711  4.184536  4.636023  4.694756 -2.214753   \n",
       "C3N-01651   4.241089  4.465306  4.823650  4.909012  4.214024 -0.137019   \n",
       "C3N-01808   4.476363  4.189120  4.141382  4.853221  4.731376 -1.986306   \n",
       "\n",
       "Name            MT1G      MT1F      MT1E      MT1B tumor_stage_pathological  \n",
       "Patient_ID                                                                   \n",
       "C3L-00004  -2.253525 -2.546547 -2.557182 -1.939419                Stage III  \n",
       "C3L-00010  -4.060004 -2.921551 -3.591516 -3.363261                  Stage I  \n",
       "C3L-00026  -2.475965 -2.492909 -2.004049 -1.809835                  Stage I  \n",
       "C3L-00079  -2.525792 -3.036604 -2.514611 -1.763428                Stage III  \n",
       "C3L-00088  -2.110608 -1.972565 -2.471242 -1.731799                Stage III  \n",
       "...              ...       ...       ...       ...                      ...  \n",
       "C3N-01646  -2.047085 -2.162791 -2.431460 -1.742155                Stage III  \n",
       "C3N-01648  -1.196218 -1.313544 -1.182120 -1.103926                 Stage II  \n",
       "C3N-01649  -2.534672 -2.485910 -2.172642 -2.360767                Stage III  \n",
       "C3N-01651  -0.360000 -0.295140 -3.646967 -0.165253                 Stage II  \n",
       "C3N-01808  -1.697919 -2.052849 -1.963289 -1.544815                  Stage I  \n",
       "\n",
       "[98 rows x 11 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Combining genes and proteins of interest into one dataframe\n",
    "top_ten_genes = pd.concat([top_RNA_df, top_protein_df], axis=1)\n",
    "\n",
    "#Creating separate list of the patients' cancer stages\n",
    "\n",
    "top_ten_genes[\"tumor_stage_pathological\"] = clinical_data.loc[:, \"tumor_stage_pathological\"]\n",
    "\n",
    "top_ten_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "13e28946-ab62-4593-947a-b4372a952478",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import OrdinalEncoder\n",
    "\n",
    "encoder = OrdinalEncoder()\n",
    "unencoded_columns = patient_stages\n",
    "encoded_columns = encoder.fit_transform(unencoded_columns)\n",
    "patient_stages = encoded_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "5410cc28-9240-4077-9d32-52931c84984d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import StandardScaler\n",
    "from umap import UMAP\n",
    "\n",
    "\n",
    "scaler = StandardScaler()\n",
    "\n",
    "embedding_unscaled = UMAP().fit_transform(feature_df)\n",
    "\n",
    "scaled_data = scaler.fit_transform(feature_df)\n",
    "embedding_scaled = UMAP().fit_transform(scaled_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f09abe08-8141-4ef7-9bea-75ae7dd3afcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "Testing different algorithms for best fit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "20dc44b3-bda1-454b-8e5b-3a2307a8ae78",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.neighbors import KNeighborsClassifier # default number of neighbors looked at is 5\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.naive_bayes import GaussianNB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "f6bb728d-c1e9-4f4d-aaf6-eb775fa73cf6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#single run\n",
    "\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "patient_stages = np.array(patient_stages)\n",
    "\n",
    "classifier = KNeighborsClassifier() # your choice of which classifier to use\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(scaled_data, patient_stages, train_size=0.7)\n",
    "\n",
    "classifier.fit(X_train, y_train.ravel())\n",
    "\n",
    "y_pred = classifier.predict(X_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ebc84360-b40d-4fa8-8d4d-bfcbeefb53e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "After 10 simulations, the average error for each regressor is as follows:\n",
      "\tKNeighborsClassifier : 0.21666666666666665\n",
      "\tDecisionTreeClassifier : 0.2566666666666666\n",
      "\tMLPClassifier : 0.16\n",
      "\tGaussianNB : 0.13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n",
      "/Users/akki/miniconda3/envs/qbio_490_Akansha2/lib/python3.9/site-packages/sklearn/neural_network/_multilayer_perceptron.py:691: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (200) reached and the optimization hasn't converged yet.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "regressors_names = [\n",
    "    'KNeighborsClassifier', \n",
    "    'DecisionTreeClassifier', \n",
    "    'MLPClassifier', \n",
    "    'GaussianNB'\n",
    "]\n",
    "\n",
    "regressors = [\n",
    "    KNeighborsClassifier(),\n",
    "    DecisionTreeClassifier(),\n",
    "    MLPClassifier(),\n",
    "    GaussianNB()\n",
    "]\n",
    "\n",
    "regressors_perf = {\n",
    "    0: [],\n",
    "    1: [],\n",
    "    2: [],\n",
    "    3: []\n",
    "}\n",
    "\n",
    "for n in range(10):\n",
    "    X_train, X_test, y_train, y_test = train_test_split(scaled_data, patient_stages, train_size=0.7)\n",
    "    for i in range(len(regressors)):\n",
    "        model = regressors[i]\n",
    "        model.fit(X_train, y_train.ravel())\n",
    "        y_pred = model.predict(X_test)\n",
    "\n",
    "        difference = y_pred - y_test\n",
    "        regressors_perf[i].append(abs(np.mean(difference)))\n",
    "\n",
    "print('\\nAfter 10 simulations, the average error for each regressor is as follows:')\n",
    "for i in regressors_perf:\n",
    "    print(f'\\t{regressors_names[i]} : {np.mean(regressors_perf[i])}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "926ff6f6-8a0f-4959-84b6-318f75f1db7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#GaussianNB appears to be the best model as it has the least error. "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
